src/avfoundation.cs

```diff
@@ -3183,6 +3194,109 @@ interface AVCaptureSynchronizedDepthData
+	interface AVQueuedSampleBufferRendering
+	{
+		[Abstract]
+		[Export ("timebase", ArgumentSemantic.Retain)]
+		CMTimebase Timebase { get; }
+
+		[Abstract]
+		[Export ("enqueueSampleBuffer:")]
+		void Enqueue (CMSampleBuffer sampleBuffer);
+
+		[Abstract]
+		[Export ("flush")]
+		void Flush ();
+
+		[Abstract]
+		[Export ("readyForMoreMediaData")]
+		bool ReadyForMoreMediaData { [Bind ("isReadyForMoreMediaData")] get; }
+
+		[Abstract]
+		[Export ("requestMediaDataWhenReadyOnQueue:usingBlock:")]
+		void RequestMediaData (DispatchQueue queue, Action handler);
+
+		[Abstract]
+		[Export ("stopRequestingMediaData")]
+		void StopRequestingMediaData ();
+	}
+
+	[TV (11,0), NoWatch, Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface AVSampleBufferAudioRenderer : AVQueuedSampleBufferRendering
+	{
+		[Export ("status")]
+		AVQueuedSampleBufferRenderingStatus Status { get; }
+
+		[NullAllowed, Export ("error")]
+		NSError Error { get; }
+
+		[NullAllowed, Export ("audioOutputDeviceUniqueID")]
+		string UniqueId { get; set; }
+
+		[Export ("audioTimePitchAlgorithm")]
+		string PitchAlgorithm { get; set; }
+
+		// AVSampleBufferAudioRenderer_AVSampleBufferAudioRendererVolumeControl
+		[Export ("volume")]
+		float Volume { get; set; }
+
+		[Export ("muted")]
+		bool Muted { [Bind ("isMuted")] get; set; }
+
+		// AVSampleBufferAudioRenderer_AVSampleBufferAudioRendererQueueManagement
+
+		[Async]
+		[Export ("flushFromSourceTime:completionHandler:")]
+		void Flush (CMTime time, Action<bool> completionHandler);
+	}
+
+	interface IAVQueuedSampleBufferRendering {}
+
+	[TV (11,0), NoWatch, Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface AVSampleBufferRenderSynchronizer
+	{
+		[Export ("timebase", ArgumentSemantic.Retain)]
+		CMTimebase Timebase { get; }
+
+		[Export ("rate")]
+		float Rate { get; set; }
+
+		[Export ("setRate:time:")]
+		void SetRate (float rate, CMTime time);
+
+		// AVSampleBufferRenderSynchronizer_AVSampleBufferRenderSynchronizerRendererManagement
+
+		[Export ("renderers")]
+		IAVQueuedSampleBufferRendering[] Renderers { get; }
+
+		[Export ("addRenderer:")]
+		void Add (IAVQueuedSampleBufferRendering renderer);
+
+		[Async]
+		[Export ("removeRenderer:atTime:completionHandler:")]
+		void Remove (IAVQueuedSampleBufferRendering renderer, CMTime time, [NullAllowed] Action<bool> completionHandler);
+
+		// AVSampleBufferRenderSynchronizer_AVSampleBufferRenderSynchronizerTimeObservation
+
+		// as per the docs the returned observers are an opaque object that you pass as the argument to 
+		// removeTimeObserver to cancel observation.
+
+		[Async]
+		[Export ("addPeriodicTimeObserverForInterval:queue:usingBlock:")]
+		NSObject AddPeriodicTimeObserver (CMTime interval, [NullAllowed] DispatchQueue queue, Action<CMTime> handler);
+
+		[Export ("addBoundaryTimeObserverForTimes:queue:usingBlock:")]
+		NSObject AddBoundaryTimeObserver (NSValue[] times, [NullAllowed] DispatchQueue queue, Action handler);
```

Comment:
`[Async]` ?

Reply:
This was reviewed in PR https://github.com/xamarin/xamarin-macios/pull/2694 ... I don't know why we missed it there. Adding it.

Reply:
The delegate can be called back multiple times (once for each value in the `times` array according to the documentation), which makes it a bad fit for `[Async]` (and the same for the method just above with the internal argument).

Reply:
Ah! Good catch, fixing. Thx @rolfbjarne

Reply:
let's add a comment here *and* the one above (and remove its `[Async]`)